openslides/core/static/js/core/base.js

```diff
@@ -69,6 +71,28 @@ angular.module('OpenSlidesApp.core', [
           console.error('The constant REALM is not set properly.');
         }
 
+        /* The callbacks re invoked if the ws connection closes and this factory tries to
+         * reconnect after 1s. The callbacks should return a promise resolveing to true/false.
+         * If resolving to true, the retry-process is stopped, so the callback can indicate
+         * whether it has done the reconnecting different.*/
+        var runRetryConnectCallbacks = function () {
```

Comment:
I do not understand the docstring.

But you can simplfy this:

```javascript
var runRetryConnectCallbacks = function () {
    var callbackPromises = _.map(retryConnectCallbacks, function (callback) {
        return callback();
    });
    $q.all(callbackPromises).then(
        function (data) {
            // All promises resolved successfully. Now try to reconnect.
            Autoupdate.newConnect();
        },
        function (error) {
            // At least one promise resolved with a rejection. Stop everything ... ???
            ErrorMessage.clearConnectionError();
        }
    );
}
```

Now you have to change the callbacks that they return a promise that resolve or reject correctly.